### DISCUSSION TITLE
How do you adjust `window_margin`?

### DISCUSSION THREAD

**Discussion by j-n-f** [emilk/egui#4365] [Q&A]
I tried to put the following in the `update` function (before anything gets drawn):

```rust
        ctx.style_mut(|style| {
            style.spacing.window_margin = egui::Margin::same(32.0);
        });
```

I also tried doing similar before the `update` function is ever called, and it still doesn't work. There's nothing in the `ViewportBuilder` that will let me achieve this either.

The only workaround is to manually set the margins for each top/bottom, side, and (the single) central panel. This seems inefficient. There is a `Frame` somewhere in the code for the window, and there must be a way to access it. I'm just not seeing it.

**Comment by radiantly** [comment]
For anyone else who's come searching for this, the workaround is to set the margins manually on each of the panels like so:
```rust
egui::CentralPanel::default()
    .frame(egui::Frame::default().inner_margin(0.0))
    .show(...)
```
